packages/fx-core/src/plugins/resource/cicd/providers/provider.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { FxError, Result, ok } from "@microsoft/teamsfx-api";
+import * as fs from "fs-extra";
+import { FileSystemError, InternalError, NoProjectOpenedError } from "../errors";
+import { TemplateKind } from "./enums";
+import path from "path";
+import Mustache from "mustache";
+import { getTemplatesFolder } from "../../../../folder";
+import { Context } from "@microsoft/teamsfx-api/build/v2";
+import { generateBuildScript } from "../utils/buildScripts";
+
+export class CICDProvider {
+  public scaffoldTo = "";
+  public providerName = "";
+  public sourceTemplateName?: (templateName: string) => string;
+  public targetTemplateName?: (templateName: string, envName: string) => string;
+  public async scaffold(
+    projectPath: string,
+    templateName: string,
+    envName: string,
+    context: Context
+  ): Promise<Result<boolean, FxError>> {
```

Comment:
Why we don't use FxResult as return type, which is defined in packages/fx-core/src/plugins/resource/cicd/result.ts?

Reply:
Updated - use FxResult type instead.